typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -181,30 +181,27 @@ runPeer
   -> Channel m bytes
   -> Peer ps pr st m a
   -> m a
-runPeer tr codec peerid channel =
-  runPeerWithDriver (driverSimple tr peerid codec channel)
+runPeer tr codec peerid channel peer =
+    fst <$> runPeerWithDriver driver peer (startDState driver)
+  where
+    driver = driverSimple tr peerid codec channel
 
 runPeerWithDriver
   :: forall ps (st :: ps) pr dstate m a.
      Monad m
   => Driver ps dstate m
   -> Peer ps pr st m a
-  -> m a
-runPeerWithDriver Driver{sendMessage, recvMessage, startDState} =
-    go startDState
+  -> dstate
```

Comment:
The using the input and output state is only needed if you want to chain two driver runs together or if you know something special about the driver. Otherwise using the empty/initial state and ignoring the output state is the thing to do. Can certainly document that.